File: binary_tree_inorder_traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    const res = [];
    helper(root, res);
    return res;
};

function helper(node, res) {
    if(!node) {
        return;
    }
    helper(node.left, res);
    res.push(node.val)
    helper(node.right, res);
}

File: problems/binaryTreeInorderTraversal.py
"""
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        
"""
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        return self.helper(root, [])

    def helper(self, node, res):
        if not node:
            return 
        
        self.helper(node.left, res)
        res.append(node.val)
        self.helper(node.right, res)
        return res
        